examples/ca-csr-example.cpp

```diff
@@ -0,0 +1,220 @@
+                .digestType(digestType)
+                .addExtension(rootCertSignConstraints)
+                .addExtension(rootCaKeyUsage)
+                .build()
+                );
+
+    return CertificateAuthority(
+                rootCaSignParams,
+                // next serial number
+                1,
+                rootCaCert,
+                rootCaKey);
+}
+
+CertificateAuthority getIntermediateCa(const X509Certificate &intermediateCaCert,
+                                       const AsymmetricKeypair &intermediateCaKey)
+{
+    auto intermediateCaKeyUsage = KeyUsageExtension::Builder{}.digitalSignature().keyEncipherment().build();
+    // Set CA = false for leaf certificates. Path length is ignored then.
+    BasicConstraintsExtension intermediateCaSigningConstraints(false, 0);
+
+    auto intermediateCaSignParams = CertificateSigningParameters::Builder{}
+            .certificateValidity(Asn1Time::Seconds(120))
+            .notBeforeAsn1(Asn1Time::now() - std::chrono::seconds{1})
+            .digestType(digestType)
+            .addExtension(intermediateCaSigningConstraints)
+            .addExtension(intermediateCaKeyUsage)
+            .build();
+
+    return CertificateAuthority(
+                intermediateCaSignParams,
+                // next serial number
+                1,
+                intermediateCaCert,
+                intermediateCaKey);
+}
+
+X509Certificate createRootCertificate(const AsymmetricKeypair &rootEccKey)
+{
+    // Set the CA to true and path length to 1 (only one intermediate CA)
+    // This will be the actual X509 content of the root CA certificate
+    BasicConstraintsExtension rootCertConstraint{true, 1};
+
+    auto rootCaSelfSignParams = CertificateSigningParameters(
+                CertificateSigningParameters::Builder{}
+                .certificateValidity(Asn1Time::Seconds(120))
+                .digestType(digestType)
+                .addExtension(rootCertConstraint)
+                .addExtension(rootCaKeyUsage)
+                .build()
+                );
+
+    // These are all supported X509 cert details supported by mococrw
+    auto rootCertDetails = DistinguishedName::Builder{}
+            .commonName("ImATeapot")
+            .countryName("DE")
+            .organizationName("Linux AG")
+            .organizationalUnitName("Linux Support")
+            .pkcs9EmailAddress("support@example.com")
+            .localityName("oben")
+            .stateOrProvinceName("nebenan")
+            .serialNumber("08E36DD501941432358AFE8256BC6EFD")
+            .givenName("Hot Teapot")
+            .userId("1000")
+            .title("Teapots contain tea")
+            .build();
+
+    return CertificateAuthority::createRootCertificate(
+                rootEccKey,
+                rootCertDetails,
+                // serial number
+                0,
+                rootCaSelfSignParams);
+}
+
+X509Certificate singCertificateWithRootCa(const X509Certificate &signingCert,
```

Comment:
```suggestion
X509Certificate signCertificateWithRootCa(const X509Certificate &signingCert,
```

We should have a certificate singing contest, that could be fun :-D

Reply:
:)